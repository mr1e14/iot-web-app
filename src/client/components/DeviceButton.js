import React from "react";
import PropTypes from "prop-types";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";
import { Link } from "react-router-dom";

const DeviceButton = props => {
  const { className, text, iconSrc, iconAlt, linkTo } = props;

  return (
    <Button
      component={Link}
      to={linkTo}
      variant="contained"
      fullWidth={true}
      className={className}
    >
      <Typography color="textPrimary">{text}</Typography>
      <img src={iconSrc} alt={iconAlt} style={{ padding: "0.25rem 0.5rem" }} />
    </Button>
  );
};

DeviceButton.propTypes = {
  className: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  iconSrc: PropTypes.string.isRequired,
  linkTo: PropTypes.string.isRequired
};

export default DeviceButton;
