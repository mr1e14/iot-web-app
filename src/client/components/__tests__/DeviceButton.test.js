import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

jest.mock("../mediaQuery");

import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import DeviceButton from "../DeviceButton";

const exampleDeviceButtonProps = {
  className: "button1",
  text: "Manage lights",
  iconSrc: "../img/some_img.png",
  iconAlt: "lights-button"
};

describe("DeviceButton", () => {
  let tree;
  let component;
  describe("when props are supplied", () => {
    beforeEach(() => {
      component = renderer.create(
        <DeviceButton {...exampleDeviceButtonProps} />
      );
      tree = component.toJSON();
    });
    it("renders correctly", () => {
      expect(tree).toMatchSnapshot();
    });
  });
  describe("when renders", () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(<DeviceButton {...exampleDeviceButtonProps} />);
    });
    it("has one Button", () => {
      expect(wrapper.find(Button).length).toEqual(1);
    });
    it("has correct Button class", () => {
      expect(wrapper.find(Button).hasClass("button1")).toEqual(true);
    });
    it("displays text related to a device", () => {
      expect(wrapper.find(Typography).text()).toBe("Manage lights");
    });
    it("displays device icon", () => {
      expect(wrapper.find("img").prop("src")).toBe("../img/some_img.png");
      expect(wrapper.find("img").prop("alt")).toBe("lights-button");
    });
  });
});
