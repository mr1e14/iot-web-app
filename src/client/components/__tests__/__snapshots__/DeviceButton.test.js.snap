// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceButton when props are supplied renders correctly 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root button1 MuiButton-contained MuiButton-fullWidth"
  disabled={false}
  onBlur={[Function]}
  onDragEnd={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
    >
      Manage lights
    </p>
    <img
      alt="lights-button"
      src="../img/some_img.png"
      style={
        Object {
          "padding": "0.25rem 0.5rem",
        }
      }
    />
  </span>
</button>
`;
